acvm/src/compiler/transformers/fallback.rs

```diff
@@ -6,47 +6,50 @@ use acir::{
 };
 
 // A predicate that returns true if the black box function is supported
-pub type IsBlackBoxSupported = fn(&BlackBoxFunc) -> bool;
+pub type IsOpcodeSupported = fn(&Opcode) -> bool;
 
 pub struct FallbackTransformer;
 
 impl FallbackTransformer {
     //ACIR pass which replace unsupported opcodes using arithmetic fallback
     pub fn transform(
         acir: Circuit,
-        is_supported: IsBlackBoxSupported,
+        is_supported: IsOpcodeSupported,
     ) -> Result<Circuit, CompileError> {
         let mut acir_supported_opcodes = Vec::with_capacity(acir.opcodes.len());
 
         let mut witness_idx = acir.current_witness_index + 1;
 
         for opcode in acir.opcodes {
-            let bb_func_call = match &opcode {
-                Opcode::Arithmetic(_) | Opcode::Directive(_) | Opcode::Block(_, _) => {
-                    // directive, arithmetic expression or block are handled by acvm
+            match &opcode {
+                Opcode::Arithmetic(_)
+                | Opcode::Directive(_)
+                | Opcode::Block(_)
+                | Opcode::ROM(_)
+                | Opcode::RAM(_) => {
+                    // directive, arithmetic expression or blocks are handled by acvm
                     acir_supported_opcodes.push(opcode);
                     continue;
                 }
                 Opcode::BlackBoxFuncCall(bb_func_call) => {
                     // We know it is an black box function. Now check if it is
                     // supported by the backend. If it is supported, then we can simply
                     // collect the opcode
-                    if is_supported(&bb_func_call.name) {
+                    if is_supported(&opcode) {
                         acir_supported_opcodes.push(opcode);
                         continue;
+                    } else {
+                        // If we get here then we know that this black box function is not supported
```

Comment:
since we changed blackbox function above -- we should change mention of it here to

Reply:
I think it is ok, we are in the case of a black box function.